File: Main/150/704.cc
#include <vector>
using namespace std;

class Solution {
public:
  int search(vector<int> &nums, int target) {
    int l = 0, r = nums.size() - 1;

    while (l <= r) {
      int m = (l + r) / 2;
      if (nums[m] == target)
        return m;
      if (target < nums[m]) {
        r = m - 1;
      } else {
        l = m + 1;
      }
    }
    return -1;
  }
};

File: src/2025/rusting/704.rs
struct Solution;

impl Solution {
    pub fn search(nums: Vec<i32>, target: i32) -> i32 {
        if nums.is_empty() {
            return -1;
        }
        
        let mut l = 0;
        let mut r = nums.len() as i32 - 1;

        while l <= r {
            let m = l + (r - l) / 2;
            match nums[m as usize].cmp(&target) {
                std::cmp::Ordering::Equal => return m,
                std::cmp::Ordering::Less => l = m + 1,
                std::cmp::Ordering::Greater => r = m - 1,
            }
        }

        -1
    }
}